clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -813,34 +815,56 @@ public void handle(SyncGroupResponse syncResponse,
      */
     private RequestFuture<Void> sendFindCoordinatorRequest(Node node) {
         // initiate the group metadata request
-        log.debug("Sending FindCoordinator request to broker {}", node);
-        FindCoordinatorRequest.Builder requestBuilder =
-                new FindCoordinatorRequest.Builder(
-                        new FindCoordinatorRequestData()
-                            .setKeyType(CoordinatorType.GROUP.id())
-                            .setKey(this.rebalanceConfig.groupId));
+        log.debug("Sending FindCoordinator request to broker {} with batch={}", node, batchFindCoordinator);
+        FindCoordinatorRequestData data = new FindCoordinatorRequestData()
+                .setKeyType(CoordinatorType.GROUP.id());
+        if (batchFindCoordinator) {
+            data.setCoordinatorKeys(Collections.singletonList(this.rebalanceConfig.groupId));
+        } else {
+            data.setKey(this.rebalanceConfig.groupId);
+        }
+        FindCoordinatorRequest.Builder requestBuilder = new FindCoordinatorRequest.Builder(data);
         return client.send(node, requestBuilder)
-                .compose(new FindCoordinatorResponseHandler());
+                .compose(new FindCoordinatorResponseHandler(batchFindCoordinator));
     }
 
     private class FindCoordinatorResponseHandler extends RequestFutureAdapter<ClientResponse, Void> {
+        private boolean batch;
+        FindCoordinatorResponseHandler(boolean batch) {
+            this.batch = batch;
+        }
 
         @Override
         public void onSuccess(ClientResponse resp, RequestFuture<Void> future) {
             log.debug("Received FindCoordinator response {}", resp);
 
             FindCoordinatorResponse findCoordinatorResponse = (FindCoordinatorResponse) resp.responseBody();
-            Errors error = findCoordinatorResponse.error();
+            if (batch && findCoordinatorResponse.data().coordinators().size() != 1) {
+                log.error("Group coordinator lookup failed: Invalid response");
+                future.raise(new IllegalStateException("Group coordinator lookup failed: Invalid response"));
```

Comment:
Could we have a more descriptive error message/exception here? Something more about how any coordinator size that's not 1 is unexpected since we are only looking up the coordinator for one group.